src/serialization.cpp

```diff
@@ -240,8 +348,20 @@ void compress(const SharedBuffer<u8> &data, std::ostream &os, u8 version)
 	os.write((char*)&current_byte, 1);
 }
 
+void compress(const std::string &data, std::ostream &os, u8 version, int level)
+{
+	SharedBuffer<u8> databuf((u8*)data.c_str(), data.size());
+	compress(databuf, os, version, level);
```

Comment:
It'd be better to move the implementation into here and make `compress(const SharedBuffer<u8>&, ...)` call this one because it'd avoid needlessly copying the buffer. (the `SharedBuffer` constructor does that)

Reply:
Agreed. I followed what we had for zlib. Would have to change that as well. Lemme take a stab at it.

Reply:
Won't I then have the reverse problem: Creating a string object, which makes a copy?
At best there could be a version that take a char[] and size, and both the string and sharedbuffer version call that one.

Reply:
Did that... New push coming soon.